src/EditorFeatures/Core/InlineHints/InlineHintDataTag.cs

```diff
@@ -2,22 +2,70 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System;
+using System.Linq;
 using Microsoft.CodeAnalysis.InlineHints;
+using Microsoft.CodeAnalysis.Text.Shared.Extensions;
+using Microsoft.VisualStudio.Text;
 using Microsoft.VisualStudio.Text.Tagging;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Editor.InlineHints
 {
     /// <summary>
     /// The simple tag that only holds information regarding the associated parameter name
     /// for the argument
     /// </summary>
-    internal class InlineHintDataTag : ITag
+    internal sealed class InlineHintDataTag : ITag, IEquatable<InlineHintDataTag>
     {
+        private readonly InlineHintsDataTaggerProvider _provider;
+
+        /// <summary>
+        /// The snapshot this tag was created against.
+        /// </summary>
+        private readonly ITextSnapshot _snapshot;
+
         public readonly InlineHint Hint;
 
-        public InlineHintDataTag(InlineHint hint)
+        public InlineHintDataTag(InlineHintsDataTaggerProvider provider, ITextSnapshot snapshot, InlineHint hint)
         {
+            _provider = provider;
+            _snapshot = snapshot;
             Hint = hint;
         }
+
+        public override int GetHashCode()
+            => throw ExceptionUtilities.Unreachable();
+
+        public override bool Equals(object? obj)
+            => obj is InlineHintDataTag tag && Equals(tag);
+
+        public bool Equals(InlineHintDataTag? other)
+        {
+            if (other is null)
+                return false;
+
+            // they have to match if they're going to change text.
```

Comment:
Was there a reason this was broken out versus a large && block like we usually do?